stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -0,0 +1,386 @@
+        }
+        yaw_values.push_back(0.0); //No rotation from last point
+
+        //get target time from speed
+        std::vector<double> target_times;
+        std::vector<lanelet::BasicPoint2d> trajectory_locations;
+        std::vector<double> trajectory_speeds;
+        //split point speed pair
+        splitPointSpeedPairs(future_points,&trajectory_locations,&trajectory_speeds);
+        std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(trajectory_locations);
+
+        //get trajectory time from distance and speed
+        speed_to_time(downtracks, trajectory_speeds, &target_times, jerk_);
+        std::vector <cav_msgs::TrajectoryPlanPoint> traj;
+        ros::Time start_time = ros::Time::now();
+        cav_msgs::TrajectoryPlanPoint traj_prev;
+        traj_prev.x= future_points[0].point.x();
+        traj_prev.y = future_points[0].point.y();
+        traj_prev.yaw = yaw_values[0];
+        traj_prev.target_time = start_time;
+        double delta_time = maneuver_time_/future_points.size();
+
+        for (size_t i=0; i < future_points.size(); i++)
+        {
+            cav_msgs::TrajectoryPlanPoint traj_point;
+            if(trajectory_speeds[i] > 0){
+                traj_point.x = future_points[i].point.x();
+                traj_point.y = future_points[i].point.y();
+                traj_point.yaw = yaw_values[i];
+                traj_point.target_time = start_time + ros::Duration(target_times[i]);
+            }
+            else    //speed_to_time doesn't work for 0.0 speed
+            {
+                traj_point.x=traj_prev.x;
+                traj_point.y=traj_prev.y;
+                traj_point.yaw=traj_prev.yaw;
+                traj_point.target_time = traj_prev.target_time + ros::Duration(0.1);
+                
+            }
+            traj_point.controller_plugin_name = "default";
+            traj_point.planner_plugin_name =plugin_discovery_msg_.name;
+            traj.push_back(traj_point);
+            traj_prev = traj_point;
+        }
+        
+        return traj;
+    }
+
+    void StopandWait::speed_to_time(const std::vector<double>& downtrack, const std::vector<double>& speeds, std::vector<double>* times, double jerk)
+    {
+        if(downtrack.size() !=speeds.size())
+        {
+            throw std::invalid_argument("Input vector sizes do not match");
+        }
+        if (downtrack.size() == 0)
+        {
+            throw std::invalid_argument("Input vectors are empty");
+        }
+
+        times->reserve(downtrack.size());  
+
+        //Uses equation 
+        //d_t = sqrt(2(d*v)/j)
+        double prev_speed = speeds[0];
+        double prev_time = 0.0;
+        double prev_pos = downtrack[0];
+        times->push_back(prev_time);
+        for(int i=1; i <downtrack.size();i++)
+        {
+            double cur_speed = speeds[i];
+            double delta_v = std::abs(cur_speed - prev_speed);
+            double dt = pow((2*delta_v/jerk),0.5);
```

Comment:
Doesn't matter a huge deal, but if I recall correctly the `sqrt()` function is substantially faster (and guaranteed to round correctly I believe) than `pow(..., 0.5)` for what it's worth.